rotkehlchen/chain/ethereum/modules/curve/curve_cache.py

```diff
@@ -345,3 +354,22 @@ def query_curve_meta_pools(
         pools_mapping[pool_addr] = (pool_addr, pool_coins, None)
 
     return pools_mapping
+
+
+def query_curve_gauges(
+        ethereum: 'EthereumInquirer',
+        registry_contract: 'EvmContract',
+        known_pools: list[ChecksumEvmAddress],
+) -> dict[ChecksumEvmAddress, ChecksumEvmAddress]:
+    """Queries curve gauges and returns a mapping pool address -> gauge address"""
+    pools_to_gauges: dict[ChecksumEvmAddress, ChecksumEvmAddress] = {}
+    for pool_address in known_pools:
+        gauge_address = registry_contract.call(
+            node_inquirer=ethereum,
+            method_name='get_gauges',
```

Comment:
There is a little confusion. There is always only one gauge per pool https://curve.readthedocs.io/dao-gauges.html#liquidity-gauges
![image](https://user-images.githubusercontent.com/63492346/227266112-15227452-b31a-4229-920e-1ced5473ae13.png)
 
However this particular contract method can return multiple addresses. I am not sure why it is designed like that, but with samples that I tested with the only returned address was the first address, all other were empty.

Reply:
Judging the world by 1-2 tests you make is a guarantee to write wrong code and introduce serious bugs that will take many working days to solve. Please don't ever conclude "I don't know why they do that, but I will just assume it only ever returns a single gauge".

If you want to do that then prove to me and understand why that is. Which means studyingthe code and asking in their discord/telegram whatever they have.

If you read the [code](https://etherscan.io/address/0x90E00ACe148ca3b23Ac1bC8C240C2a7Dd9c2d7f5#code).

You immediately see it's multiple gauges:

```vyper
@view
@external
def get_gauges(_pool: address) -> (address[10], int128[10]):
    """
    @notice Get a list of LiquidityGauge contracts associated with a pool
    @param _pool Pool address
    @return address[10] of gauge addresses, int128[10] of gauge types
    """
    liquidity_gauges: address[10] = empty(address[10])
    gauge_types: int128[10] = empty(int128[10])
    gauge_controller: address = self.gauge_controller
    for i in range(10):
        gauge: address = self.liquidity_gauges[_pool][i]
        if gauge == ZERO_ADDRESS:
            break
        liquidity_gauges[i] = gauge
        gauge_types[i] = GaugeController(gauge_controller).gauge_types(gauge)

    return liquidity_gauges, gauge_types
```


But then reading the code and the docs you see that it's a for loop, of 10 possible gauges per pool. And the reason it's  10 is because there is multiple liquidity gauge types: https://curve.readthedocs.io/dao-gauges.html#gauge-types

And while these are mostly other chains it's also a differentiation of crypto pools and stable pools in ethereum.

Reply:
I did a more thorough check and checked this method against all our known curve pools. There were different types of gauges returned, however there was always a single gauge (or no gauges) returned. Also if we consider the possibility that there are multiple gauges returned, how do you think we should handle it?

Reply:
We handle it by keeping a mapping of one to many if it's possible. The point is to find if it is possible by asking their engineers. As I said ... manual experimenting is pointless here. Can't sample your way into what is possible. You need to ask and get more information.

If multiple pool gauges are possible then we should keep a mapping of pool -> list of gauges, and take that into account in decoding. Doesn't that make sense to you?

Reply:
For this let's keep it like this for now but let's add more information in a subsequent PR. I will add it in the issue.